.circleci/config.yml

```diff
@@ -630,7 +627,7 @@ jobs:
                 -DPIKA_WITH_EXAMPLES=On \
                 -DPIKA_WITH_GENERIC_CONTEXT_COROUTINES=On \
                 -DPIKA_WITH_TESTS=On \
-                -DPIKA_WITH_TESTS_HEADERS=On \
+                -DPIKA_WITH_TESTS_HEADERS=Off \
```

Comment:
Does this take too long to run? It'd be nice to have it enabled, but can be enabled later e.g. on the M1 runner later as well.